browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc

```diff
@@ -188,3 +189,36 @@ TEST_F(BraveAdBlockTPNetworkDelegateHelperTest, SimpleBlocking) {
   // made (`browser_context` is `nullptr`).
   EXPECT_EQ(0ULL, host_resolver_->num_resolve());
 }
+
+TEST_F(BraveAdBlockTPNetworkDelegateHelperTest, RedirectUrl) {
+  ResetAdblockInstance(
+      g_brave_browser_process->ad_block_service(),
+      "||brave.com/test.js$redirect-url=https://test.com/test.js", "", true);
+
+  const GURL url("https://brave.com/test.js");
+  auto request_info = std::make_shared<brave::BraveRequestInfo>(url);
+  request_info->request_identifier = 1;
+  request_info->resource_type = blink::mojom::ResourceType::kScript;
+  request_info->initiator_url = GURL("https://brave.com");
+
+  EXPECT_TRUE(CheckRequest(request_info));
+  EXPECT_EQ(request_info->blocked_by, brave::kAdBlocked);
+  EXPECT_EQ("https://test.com/test.js", request_info->new_url_spec);
+}
+
+TEST_F(BraveAdBlockTPNetworkDelegateHelperTest, RedirectUrlNoHttps) {
+  ResetAdblockInstance(
+      g_brave_browser_process->ad_block_service(),
+      "||brave.com/test.js$redirect-url=http://test.com/test.js", "", true);
+
+  const GURL url("https://brave.com/test.js");
+  auto request_info = std::make_shared<brave::BraveRequestInfo>(url);
+  request_info->request_identifier = 1;
+  request_info->resource_type = blink::mojom::ResourceType::kScript;
+  request_info->initiator_url = GURL("https://brave.com");
+
+  EXPECT_TRUE(CheckRequest(request_info));
+  EXPECT_TRUE(request_info->new_url_spec.empty());
+  // EXPECT_EQ(request_info->blocked_by, brave::kNotBlocked); // TODO uncomment
+  // once https://github.com/brave/adblock-rust/pull/191 merged in
```

Comment:
The check for `https` redirect URLs should happen browser-side, so this test shouldn't depend on that PR